test/Microsoft.AspNetCore.Diagnostics.Tests/ExceptionHandlerTest.cs

```diff
@@ -176,6 +176,48 @@ public class ExceptionHandlerTest
         }
 
         [Fact]
+        public async Task Reexecute_RequestWithQueryString()
+        {
+            var expectedStatusCode = 432;
+            var destination = "/location";
+            var builder = new WebHostBuilder()
+                .Configure(app =>
+                {
+                    app.UseStatusCodePagesWithReExecute("/errorPage", "?id={0}");
+
+                    app.Map(destination, (innerAppBuilder) =>
+                    {
+                        innerAppBuilder.Run(async (httpContext) =>
+                        {
+                            await Task.Delay(1);
+                            httpContext.Response.StatusCode = expectedStatusCode;
+                        });
+                    });
+
+                    app.Map("/errorPage", (innerAppBuilder) =>
+                    {
+                        innerAppBuilder.Run(async (httpContext) =>
+                        {
+                            await httpContext.Response.WriteAsync(httpContext.Request.QueryString.Value);
+                        });
+                    });
+
+                    app.Run((context) =>
+                    {
+                        throw new InvalidOperationException("Invalid input provided.");
+                    });
+                });
+
+            using (var server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                var response = await client.GetAsync(destination);
+                Assert.Equal($"?id={expectedStatusCode}", await response.Content.ReadAsStringAsync());
+                Assert.Equal("", response.RequestMessage.RequestUri.Query);
```

Comment:
Put a Use middleware at the front of the app that verifies the query after next.